### PR TITLE
Feature/127 pdf scraper

### PR DESCRIPTION
**Pull request by elioverhoef**
Mostly documentation fixes in docstrings, some print improvements, some general bugfixes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -158,3 +158,4 @@ scraping/orphan_initial_dec.txt
 *.json
 *.pdf
 *.xml
+tests/pdfscraper_tests/test_annex_parse_data/test_annex_parse_data_pdf_parser.json
```
Comment (matthboers) [inline .gitignore:161]: *.json should include this, I hope?

File: scraping/combiner/__main__.py [hunk 2]
```diff
@@ -5,15 +5,15 @@
...
-
-class filetype(Enum):
-    decision = "decision"
-    annex = "annex"
-    epar = "epar"
-    omar = "omar"
-    ec_site = "ec_site"
-    ema_site = "ema_site"
-    excel = "excel"
+# TODO: Fix error: Enum is not defined
```
Comment (matthboers) [inline scraping/combiner/__main__.py:8]: from enum import Enum (MedicineType in ec_scraper.py)

File: scraping/filter/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- html (if the content of the PDF files is actually an HTML file)
+- unknown (if the file can't be opened in utf8 format for the HTML check)
+- wrong_doctype (if the PDF file is a different type than the filename suggests)
+
+# Input & Output
+## Input
+Location of the data folder with medicines
+## Output
+- filter.txt containing the following line for each incorrect PDF file:
+  - "{name of PDF}@{error type}@{URL}" for EPAR, OMAR and Annex files
```
Comment (matthboers) [inline scraping/filter/README.md:16]: Possibly nice to wrap these lines in \`\` to make them monospaced

**Comment by matthboers** [comment]
Should `filter.txt` be in `.gitignore`?

File: scraping/pdf_module/README.md [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
+The filepath of the file that needs to be parsed.
+
 ## Output
-- XML bestand voor elk PDF bestand in zelfde map als PDF bestand.
-- JSON bestand voor elk medicijn, met alle relevante attributen van dat medicijn.
-  - JSON naam format: {EU_number}_pdf_parser.json
+- An XML file for every PDF file, which contains all the information from the PDFs but
+  seperated into headers, sections and paragraphs.
+- A JSON file for every medicine, containing all relevant attributes of that medicine.
+  - JSON name format: {EU_number}_pdf_parser.json
```
Comment (matthboers) [inline scraping/pdf_module/README.md:27]: Wrap in code block as well? (\`\`)

File: scraping/pdf_module/docs/sphinx-source/README.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # Algemene functie
```
Comment (matthboers) [inline scraping/pdf_module/docs/sphinx-source/README.md:1]: Document is in Dutch.
Also doubting if this should be `.gitngored`.

File: scraping/pdf_module/pdf_scraper/parsers/omar_parse.py [hunk 6]
```diff
@@ -120,10 +120,10 @@ def get_alternative_treatments(bullet_points: list[str]) -> str:
 def get_significant_benefit(bullet_points: list[str], alternative_treatment: str) -> str:
     """
     This function parses out the significant benefit of the OMAR, 
-    the result is influenced by the result of a previous attribute.
+        the result is influenced by the result of a previous attribute.
```
Comment (matthboers) [inline scraping/pdf_module/pdf_scraper/parsers/omar_parse.py:123]: Doubting if this tab should be here

**Comment by matthboers** [comment]
There are a lot of files and changes from webscraper in this pull request. Try to avoid it next time.

**Review by matthboers — APPROVED** [review]
Fixes suggested are minor. Does not influence the code, but is nice to have fixed. 
Just make sure to not include code from other branches in these pull requests.

File: scraping/combiner/__main__.py (hunk 2 above)
Comment (elioverhoef) [reply scraping/combiner/__main__.py:8]: Niet belangrijk voor main nu

File: scraping/pdf_module/README.md [hunk 7]
```diff
@@ -1,23 +1,34 @@
...
+  - JSON name format: `{EU_number}_pdf_parser.json`
+
+# Other points of interest
+Parsing attributes from the XML files is multitudes faster than parsing directly
+from a PDF file. Logically it will take some time to create the XML files, but 
+they only have to be created once, which will save time in the long run.
 
-# Overige belangrijke punten
-Het parsen vanuit XML files is sneller dan vanuit PDF files door het snellere lezen.
-Het kost wat tijd om de XML files te maken, maar dit hoeft maar één keer.
```
Comment (Yfkesmit) [inline scraping/pdf_module/README.md:23]: misschien ook copyright statement toevoegen

**Review by Yfkesmit — APPROVED** [review]
priem, had alleen comments op documentatie dingen dus dat nog even verwerken maar dat geeft natuurlijk geen issues in dev

File: scraping/filter/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+Location of the data folder with medicines
+## Output
+- filter.txt containing the following line for each incorrect PDF file:
+  - `{name of PDF}@{error type}@{URL}` for EPAR, OMAR and Annex files
+  - `{name of PDF}@{error type}@{EU number}@{Brand name}@{url}` for a decision PDF
+- data folder without the incorrect PDF files
+
+# Other points of interest
+After one run, all incorrect PDF files will be removed.
+Therefore, filter.txt will be empty after two runs.
```
[the remark this answers is not in the document]
Comment (elioverhoef) [reply scraping/filter/README.md:22]: Is goed zo, was puur voor als je filter direct aanroept

File: scraping/filter/docs/sphinx-source/README.md [hunk 9]
```diff
@@ -1,22 +1,23 @@
...
+  - `{name of PDF}@{error type}@{URL}` for EPAR, OMAR and Annex files
+  - `{name of PDF}@{error type}@{EU number}@{Brand name}@{url}` for a decision PDF
+- data folder without the incorrect PDF files
 
-# Overige belangrijke punten
-Na een keer filter uitvoeren zijn alle incorrecte bestanden verwijderd, 
-na twee keer uitvoeren zal filter.txt dus ook leeg zijn.
\ No newline at end of file
+# Other points of interest
+After one run, all incorrect PDF files will be removed.
```
[the remark this answers is not in the document]
Comment (elioverhoef) [reply scraping/filter/docs/sphinx-source/README.md:21]: Helaas niet nee, handmatige kopietjes zijn nu nog nodig

**Pull request merged**